### ISSUE TITLE
Docbook/reStructuredText: xml:id / title issue (underscores, dashes)

### ISSUE THREAD

**Issue by windsurfer1122** [jgm/pandoc#7593]
For PyPI.org I convert a DocBook into a single reStructuredText with pandoc.
With the resulting reStructuredText I get `malformed hyperlink target` warnings in editors (e.g. VS Code), when using the checker for PyPI.org (plain reStructuredText, no Sphinx extensions) and failures when uploading to PyPI.org.

I found out that the syntax warnings are caused by the double underscore (`__`) in the resulting reStructuredText which pandoc created from the leading underscore (`_`) in the Docbook XML, and that pandoc doesn't recognize the xml:id as the title due to spaces are being replaced with underscores (`_`) and not dashes (`-`).
If I remove the leading underscore and convert the underscores for spaces into dashes, then pandoc recognizes the `xml:id` as the `<title>` and does not create additional hyperlink targets which cause that issue.

I couldn't find any DocBook specifications for [`xml:id`](https://tdg.docbook.org/tdg/5.1/ref-elements.html#common.attributes) what format is the correct one.
The only specs for `xml:id` I could find are from [W3.org](https://www.w3.org/TR/xml-id/) (normalized values #20, [NCname](https://www.w3.org/TR/xmlschema-2/#NCName) [[2]](https://www.w3.org/TR/1999/REC-xml-names-19990114/#NT-NCName)) , that show that a leading underscore is allowed, although not necessary if the first character is a letter, but didn't help what should be used for replacing spaces in DocBook xml:ids.
So either pandoc is doing something "wrong" or the DocBook specs just have a gap.
Maybe there are some other pandoc options that would help in my case.

That's why I post this issue here to get more information from the community and to document this interoperability issue plus my workaround (see below).
Thanks for reading and any information is heartly welcome.

**Example DocBook:**
All xml:ids in the `<section>` tags have a leading underscore and also use underscore to replace the spaces of the `<title>`.
```
<?xml version="1.0" encoding="UTF-8"?>
<article xmlns="http://docbook.org/ns/docbook" xmlns:xl="http://www.w3.org/1999/xlink" version="5.0" xml:lang="en">
<info>
<title>My Great Library</title>
<date>2021-09-22</date>
</info>
<section xml:id="_installation_of_my_great_library">
<title>Installation of My Great Library</title>
<simpara>It&#8217;s funky, just install it.</simpara>
</section>
<section xml:id="_credits">
<title>Credits</title>
<simpara>Me, myself and I.</simpara>
</section>
</article>
```

**reStructuredText result of `pandoc -f docbook -t rst -s --toc -o README.rst README.xml`:**
All xml:ids are converted to additional hyperlink targets with an additional underscore (so double), leading to reStructuredText syntax warnings and failed usage on PyPI.org (conversion to HTML).
```
================
My Great Library
================

:Date:   2021-09-22

.. contents::
   :depth: 3
..

.. __installation_of_my_great_library:

Installation of My Great Library
================================

It’s funky, just install it.

.. __credits:

Credits
=======

Me, myself and I.
```

**My workaround:**
"Fix" the xml:ids in each section tag with GNU awk.
```
gawk -i inplace -e '
{
  if (match($0, /xml:id="[^"]*"/, result)) {
    sub("=\"_", "=\"", result[0])
    gsub("_", "-", result[0])
    print substr($0, 1, RSTART-1) result[0] substr($0, RSTART+RLENGTH)
  } else {
    print $0
  }
}
' README.xml
```

"Fixed" DocBook:
```
<?xml version="1.0" encoding="UTF-8"?>
<article xmlns="http://docbook.org/ns/docbook" xmlns:xl="http://www.w3.org/1999/xlink" version="5.0" xml:lang="en">
<info>
<title>My Great Library</title>
<date>2021-09-22</date>
</info>
<section xml:id="installation-of-my-great-library">
<title>Installation of My Great Library</title>
<simpara>It&#8217;s funky, just install it.</simpara>
</section>
<section xml:id="credits">
<title>Credits</title>
<simpara>Me, myself and I.</simpara>
</section>
</article>
```

"Fixed" reStructuredText result of `pandoc -f docbook -t rst -s --toc -o README.rst README.xml`:
No additional hyperlink targets as xml:ids have been recognized as the `<title>`.
```
================
My Great Library
================

:Date:   2021-09-22

.. contents::
   :depth: 3
..

Installation of My Great Library
================================

It’s funky, just install it.

Credits
=======

Me, myself and I.
```

**Comment by jgm** [comment]
Here's the problem.

```
.. _foo:
```

is a reStructuredText link anchor with the identifier `foo`.  When the identifier already starts with an underscore, we get a doubled underscore here.

I'm not sure rST allows identifiers beginning with `_`.  I've tried
```
.. _`_foo`:
```
and
```
.. _\_foo:
```
and neither causes an error, but these are both treated as link targets with identifier `foo` rather than `_foo`.

Looking into this futher, I found https://docutils.sourceforge.io/docs/ref/rst/directives.html#identifier-normalization

So, maybe the best thing for us to do in pandoc is use the form
```
.. _`_foo`:
```
for identifiers beginning with `_`. At least then we won't get an error.  And we can let rst tools normalize these identifiers as they normally do.
